src/lualib/StringReplaceAll.ts

```diff
@@ -0,0 +1,40 @@
+function __TS__StringReplaceAll(
+    this: void,
+    source: string,
+    searchValue: string,
+    replaceValue: string | ((match: string, offset: number, string: string) => string)
+): string {
+    let replacer: (match: string, offset: number, string: string) => string;
+    if (typeof replaceValue === "string") {
+        replacer = () => replaceValue;
+    } else {
+        replacer = replaceValue;
+    }
+    const parts: string[] = [];
+    let partsIndex = 1;
+
+    const sub = string.sub;
```

Comment:
Why are you aliasing this function?

Reply:
It is for localization (optimization technique), but this is *doing it wrong*... The localization should be done at top-level scope, not within the function. This is one of the reasons why I wanted locals at top-level, btw.

Reply:
As seen in [my snippet](https://github.com/TypeScriptToLua/TypeScriptToLua/issues/1121#issuecomment-913183505), localization is done with a workaround, that is by wrapping "top-level" code inside `{}`.
If you are not familiar with this technique, here're some links I collected (since I didn't make a proposal yet):

For more details, click [here](https://stackoverflow.com/q/32945039), [here](http://lua-users.org/wiki/OptimisingUsingLocalVariables), and [here for benchmarks](https://gitspartv.github.io/LuaJIT-Benchmarks/#test1) (particularly relevant tests are: 1st & 2nd). This is optimization is very common and it is universal.

Reply:
fair enough, seems like a valid optimization

Reply:
I wanted to do the localization at the top level as well, but since lualib is all global this was hard to do. I figured putting it at the start of the function was the next best thing. There might be a way to do it using a code block like Cheatoid mentioned but it makes everything a bit ugly.

If we ever make lualib not all global, then we can move this to top level